src/test/scala/org/kibanaLoadTest/helpers/KbnClient.scala

```diff
@@ -127,36 +134,43 @@ class KbnClient(config: KibanaConfiguration) {
     logger.info(
       s"Deleting ${savedObjects.length} saved objects from [${archivePath.toString}]"
     )
-    // login to Kibana once and reuse cookie for concurrent requests
-    val cookie = this.getCookie()
-    // Allow 20 requests to be executing at once
+    // Allow specific amount of requests to be executing at once
     implicit val executionContext = ExecutionContext.fromExecutor(
       new java.util.concurrent.ForkJoinPool(
-        20
+        MAX_CONCURRENT_CONNECTIONS
       )
     )
 
     val requestsFuture = Future.sequence(
       savedObjects.map(so =>
         Future {
-          val client = HttpClientBuilder.create.build
+          var response: CloseableHttpResponse = null
+          val url =
+            s"${config.baseUrl}/api/saved_objects/${so.soType}/${so.id}?force=true"
           try {
-            val url =
-              config.baseUrl + s"/api/saved_objects/${so.soType}/${so.id}?force=true"
             val deleteRequest = new HttpDelete(url)
             deleteRequest.addHeader("Connection", "keep-alive")
             deleteRequest.addHeader("kbn-version", config.buildVersion)
-            deleteRequest.addHeader("Cookie", cookie)
-            val response = client.execute(deleteRequest)
+            response = client.execute(deleteRequest)
             response.getStatusLine
-          } finally client.close()
+          } catch {
+            case e: Throwable =>
+              throw new RuntimeException(
+                s"[$url] SavedObject deletion failed",
+                e
+              )
+          } finally {
+            if (response != null) response.close()
+          }
         }
       )
     )
 
     requestsFuture.onComplete {
       case Success(res) =>
-        val successCount = res.count(r => r.getStatusCode == 200)
+        client.close()
```

Comment:
IMHO the control flow in this method is already too complex to do proper resource management. I only had a quick look but you already leak `client` and `connManager` when `checkFilesExist` raises an exception. I suggest to move the code into a dedicated method and only keep the resource handling here. Something along the lines of:

```scala
  def unload(path: Path): Unit = {
    val (client, connManager) = getClientAndConnectionManager()
    try {
        doUnload(path, client, connManager)
    } finally {
        client.close()
        connManager.close();
    }
}
```

Or better yet, use [`scala.util.Using`](https://www.scala-lang.org/api/current/scala/util/Using$.html).